istioctl/cmd/injector-list.go

```diff
@@ -0,0 +1,328 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"sort"
+	"strconv"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/spf13/cobra"
+	admit_v1 "k8s.io/api/admissionregistration/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	api_pkg_labels "k8s.io/apimachinery/pkg/labels"
+
+	"istio.io/api/annotation"
+	"istio.io/api/label"
+	"istio.io/istio/galley/pkg/config/analysis/analyzers/injection"
+	analyzer_util "istio.io/istio/galley/pkg/config/analysis/analyzers/util"
+	"istio.io/istio/istioctl/pkg/clioptions"
+	"istio.io/istio/pkg/config/resource"
+	"istio.io/istio/pkg/kube"
+)
+
+type revisionCount struct {
+	// pods in a revision
+	pods int
+	// pods that are disabled from injection
+	disabled int
+	// pods that are enabled for injection, but whose revision doesn't match their namespace's revision
+	needsRestart int
+}
+
+func injectorCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "injector",
+		Short:   "List sidecar injector and sidecar versions",
+		Long:    `List sidecar injector and sidecar versions`,
+		Example: `  istioctl experimental injector list`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.HelpFunc()(cmd, args)
+			if len(args) != 0 {
+				return fmt.Errorf("unknown subcommand %q", args[0])
+			}
+
+			return nil
+		},
+	}
+
+	cmd.AddCommand(injectorListCommand())
+	return cmd
+}
+
+func injectorListCommand() *cobra.Command {
+	var opts clioptions.ControlPlaneOptions
+	cmd := &cobra.Command{
+		Use:     "list",
+		Short:   "List sidecar injector and sidecar versions",
+		Long:    `List sidecar injector and sidecar versions`,
+		Example: `  istioctl experimental injector list`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client, err := kubeClientWithRevision(kubeconfig, configContext, opts.Revision)
```

Comment:
nit: I think it would be appropriate to create a context.Background() here and pass it in to functions below, rather than using instead of using context.TODO() in multiple places.